g3doc/obtaining-training-data.md

```diff
@@ -0,0 +1,50 @@
+# Obtaining Training Data
+
+This document gives an overview of how to use the Gematria tooling to
+parse/process training data from multiple sources.
+
+## Parsing training data from the BHive dataset
+
+The publically available [BHive](https://github.com/ithemal/bhive) dataset
+has about 300k basic blocks along with throughput data for Ivy Bridge, Haswell,
+and Skylake in a CSV format. Gematria provides native tooling to process these
+CSV files into `.tfrecord` files that can be used with Gematria models.
+
+Every line in the BHive CSV files has the format
+`<machine code of BB in hex>,<throughput in cycles>>`. This is referred to as
+the BHive format. The dataset contains one CSV file per mircoarchitecture.
+To process BHive throughput data for the Intel Skylake
+microarchitecture, we can begin by downloading one of the BHive CSVs:
+
+```bash
+curl -L https://raw.githubusercontent.com/ithemal/bhive/5f1d50077ac0779fd227b261dcf517862c7104bd/benchmark/throughput/skl.csv > skl.csv
+```
+
+And then convert the downloaded CSV file into a `.tfrecord` file:
+
+```bash
+bazel run //gematria/datasets/python:import_from_bhive -- \
+    --gematria_input_csv=skl.csv \
+    --gematria_output_tfrecord=skl.tfrecord \
+    --gematria_throughput_source_name="bhive: skl"
+```
+
+This will create a `skl.tfrecord` file containing imported basic blocks that
+can be used for further downstream analysis. The `--gematria_input_csv` and
+`--gematria_output_tfrecord` flags specify the path to the input CSV and the
+path to the output `.tfrecord` files respectively. The
+`--gematria_throughput_source_name` flag should be used to specify the method
+of data collection. More information on this flag is available in the protobuf
```

Comment:
The throughput source name is there to disambiguate throughput information for different microarchitectures.

The proto data format allows storing throughput information from multiple sources (measurement methods, microarchitectures, ...) for the same basic block in the same proto. This way, we can have a single data set with throughput data for all known microarchitectures. The format (and the code around it) is flexible enough that it works even when some blocks in the data set have data only from some sources.

Typically, we'd have `bhive: skl`, `bhive: hsw`, `bhive: ivb`, ... in the same data set and train it using multi-task training.

Reply:
I added a note on this info. I was trying not to duplicate information between the protobuf definition and the documentation, but just pointing to the code from the docs is probably not a super great strategy. Thanks for the detailed explanation!